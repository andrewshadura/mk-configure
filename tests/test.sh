#!/bin/sh

env CPPFLAGS="$CPPFLAGS -I$SRCDIR" $MAKE -f $SRCDIR/tests/mkc_test.mk \
    > $OBJDIR/_test.res

diff -u $SRCDIR/tests/test.out $OBJDIR/_test.res
