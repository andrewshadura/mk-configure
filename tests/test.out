HAVE_HEADER.sys_time_h=1
HAVE_HEADER.string_h=1
HAVE_FUNCLIB.strcpy=1
HAVE_FUNCLIB.sqrt.m=1
HAVE_DEFINE.__GNUC__=1
HAVE_FUNC2.strcmp.string_h=1
HAVE_FUNC3.strcpy=0
SIZEOF.int=n
SIZEOF.long_long=n
SIZEOF.voidP=n
SIZEOF.size_t.string_h=n
HAVE_TYPE.size_t.string_h=1
HAVE_TYPE.mbstate_t.wchar_h=1
HAVE_TYPE.bad_type=0
HAVE_HEADER.bad_header_h=0
HAVE_HEADER.bad_dir_bad_header_h=0
HAVE_FUNCLIB.bad_func=0
HAVE_FUNCLIB.bad_func.bad_lib=0
HAVE_DEFINE.__BAD_DEFINE__=0
HAVE_FUNC1.bad_func=0
HAVE_FUNC1.bad_func.bad_header=0
SIZEOF.bad_type=failed
SIZEOF.bad_type.bad_header_h=failed
HAVE_FUNCLIB.superfunc1=0
HAVE_FUNCLIB.superfunc2.superlib2=0
HAVE_MEMBER.tm_tm_isdst.time_h=1
HAVE_MEMBER.sockaddr_in_sin_addr_s_addr.netinet_in_h=1
HAVE_MEMBER.bad_member=0
HAVE_MEMBER.bad_member.string_h=0
HAVE_HEADER.tests_mkc_test_h=1
HAVE_DEFINE.MKC_TEST_DEFINE.tests_mkc_test_h=1
HAVE_FUNC5.mkc_test_func.tests_mkc_test_h=1
HAVE_VAR.mkc_test_var.tests_mkc_test_h=1
HAVE_MEMBER.mkc_test_t_a.tests_mkc_test_h=1
HAVE_MEMBER.mkc_test_t_b_c.tests_mkc_test_h=1
HAVE_CUSTOM.custom_check1=1
HAVE_CUSTOM.custom_check2=0
MKC_CFLAGS=-DHAVE_HEADER_SYS_TIME_H=1 -DHAVE_HEADER_STRING_H=1 -DHAVE_HEADER_TESTS_MKC_TEST_H=1 -DSIZEOF_INT=n -DSIZEOF_LONG_LONG=n -DSIZEOF_VOIDP=n -DSIZEOF_SIZE_T_STRING_H=n -DHAVE_DEFINE___GNUC__=1 -DHAVE_DEFINE_MKC_TEST_DEFINE_TESTS_MKC_TEST_H=1 -DHAVE_TYPE_SIZE_T_STRING_H=1 -DHAVE_TYPE_MBSTATE_T_WCHAR_H=1 -DHAVE_VAR_ERRNO_ERRNO_H=1 -DHAVE_VAR_MKC_TEST_VAR_TESTS_MKC_TEST_H=1 -DHAVE_MEMBER_TM_TM_ISDST_TIME_H=1 -DHAVE_MEMBER_SOCKADDR_IN_SIN_ADDR_S_ADDR_NETINET_IN_H=1 -DHAVE_MEMBER_MKC_TEST_T_A_TESTS_MKC_TEST_H=1 -DHAVE_MEMBER_MKC_TEST_T_B_C_TESTS_MKC_TEST_H=1 -DHAVE_FUNC2_STRCMP_STRING_H=1 -DHAVE_FUNC5_MKC_TEST_FUNC_TESTS_MKC_TEST_H=1 -DHAVE_CUSTOM_CUSTOM_CHECK1=1
MKC_SRCS=superfunc1.c superfunc2.c
MKC_LDADD=-lm

KNOWN_SYSTEM
MKC_COMMON_DEFINES_WORKS_FINE
 -------------------------------------------
HAVE_HEADER.sys_time_h=2
HAVE_HEADER.string_h=2
HAVE_FUNCLIB.strcpy=2
HAVE_FUNCLIB.sqrt.m=2
HAVE_DEFINE.__GNUC__=1
HAVE_FUNC2.strcmp.string_h=2
HAVE_FUNC3.strcpy=2
SIZEOF.int=n
SIZEOF.long_long=n
SIZEOF.voidP=n
SIZEOF.size_t.string_h=n
HAVE_TYPE.size_t.string_h=1
HAVE_TYPE.mbstate_t.wchar_h=2
HAVE_TYPE.bad_type=2
HAVE_HEADER.bad_header_h=0
HAVE_HEADER.bad_dir_bad_header_h=0
HAVE_FUNCLIB.bad_func=0
HAVE_FUNCLIB.bad_func.bad_lib=0
HAVE_DEFINE.__BAD_DEFINE__=2
HAVE_FUNC1.bad_func=0
HAVE_FUNC1.bad_func.bad_header=0
SIZEOF.bad_type=failed
SIZEOF.bad_type.bad_header_h=failed
HAVE_FUNCLIB.superfunc1=0
HAVE_FUNCLIB.superfunc2.superlib2=0
HAVE_MEMBER.tm_tm_isdst.time_h=1
HAVE_MEMBER.sockaddr_in_sin_addr_s_addr.netinet_in_h=1
HAVE_MEMBER.bad_member=2
HAVE_MEMBER.bad_member.string_h=2
HAVE_HEADER.tests_mkc_test_h=1
HAVE_DEFINE.MKC_TEST_DEFINE.tests_mkc_test_h=1
HAVE_FUNC5.mkc_test_func.tests_mkc_test_h=1
HAVE_VAR.mkc_test_var.tests_mkc_test_h=2
HAVE_MEMBER.mkc_test_t_a.tests_mkc_test_h=1
HAVE_MEMBER.mkc_test_t_b_c.tests_mkc_test_h=1
HAVE_CUSTOM.custom_check1=0
HAVE_CUSTOM.custom_check2=1
MKC_CFLAGS=-DHAVE_HEADER_SYS_TIME_H=2 -DHAVE_HEADER_STRING_H=2 -DHAVE_HEADER_TESTS_MKC_TEST_H=1 -DSIZEOF_INT=n -DSIZEOF_LONG_LONG=n -DSIZEOF_VOIDP=n -DSIZEOF_SIZE_T_STRING_H=n -DHAVE_DEFINE___GNUC__=1 -DHAVE_DEFINE___BAD_DEFINE__=1 -DHAVE_DEFINE_MKC_TEST_DEFINE_TESTS_MKC_TEST_H=1 -DHAVE_TYPE_SIZE_T_STRING_H=1 -DHAVE_TYPE_MBSTATE_T_WCHAR_H=1 -DHAVE_TYPE_BAD_TYPE=1 -DHAVE_VAR_ERRNO_ERRNO_H=1 -DHAVE_VAR_MKC_TEST_VAR_TESTS_MKC_TEST_H=1 -DHAVE_MEMBER_TM_TM_ISDST_TIME_H=1 -DHAVE_MEMBER_SOCKADDR_IN_SIN_ADDR_S_ADDR_NETINET_IN_H=1 -DHAVE_MEMBER_BAD_MEMBER=1 -DHAVE_MEMBER_BAD_MEMBER_STRING_H=1 -DHAVE_MEMBER_MKC_TEST_T_A_TESTS_MKC_TEST_H=1 -DHAVE_MEMBER_MKC_TEST_T_B_C_TESTS_MKC_TEST_H=1 -DHAVE_FUNC2_STRCMP_STRING_H=1 -DHAVE_FUNC3_STRCPY=1 -DHAVE_FUNC5_MKC_TEST_FUNC_TESTS_MKC_TEST_H=1 -DHAVE_CUSTOM_CUSTOM_CHECK2=1
MKC_SRCS=superfunc1.c superfunc2.c
MKC_LDADD=-lm

KNOWN_SYSTEM
MKC_COMMON_DEFINES_WORKS_FINE
